### PR TITLE
blockchain tests fixes

### PR DESCRIPTION
**Pull request by CJentzsch**
* fix GasPricer (include all transaction not only with min gasPrice = 10 szabo)
* add ability to create blockchain tests with equal uncles headers
* fix genesis hash in blockchain tests (update after chainging PoW parameters)

### CODE REVIEW

File: test/bcUncleTestFiller.json [hunk 1]
```diff
@@ -265,23 +265,7 @@
...
-                        "nonce" : "18a524c1790fa83b",
-                        "number" : "2",
-                        "parentHash" : "6134fc6b5d99ee03c4aab1592640f6f9dcbc850668d75d631aee34989b938fae",
-                        "receiptTrie" : "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
-                        "seedHash" : "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
-                        "stateRoot" : "ff640b30d613c35dad43e3693329e1b1ee6350f989cf46a288025a1cbfdab9cd",
-                        "timestamp" : "0x54c98c82",
-                        "transactionsTrie" : "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
-                        "uncleHash" : "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347"
+                        "theSameAsBefore" : "1"
```
Comment (subtly) [inline test/bcUncleTestFiller.json]: sameAsPreviousSibling ?

File: test/blockchain.cpp [hunk 2]
```diff
@@ -101,10 +102,18 @@ void doBlockchainTests(json_spirit::mValue& _v, bool _fillin)
 				BlockQueue uncleBlockQueue;
 				mArray aUncleList;
 				vector<BlockInfo> vBiUncles;
+				mObject uncleHeaderObj_pre;
 
 				for (auto const& uHObj: blObj["uncleHeaders"].get_array())
 				{
 					mObject uncleHeaderObj = uHObj.get_obj();
+					if ( uncleHeaderObj.count("theSameAsBefore") )
```
Comment (subtly) [inline test/blockchain.cpp]: remove spaces?

File: mix/ClientModel.cpp [hunk 3]
```diff
@@ -339,18 +347,77 @@ void ClientModel::showDebuggerForTransaction(ExecutionResult const& _t)
...
+
+			//format solidity context values
+			QStringList	locals;
+			for(auto l: solLocals)
+				if (l.first < (int)s.stack.size())
+					locals.push_back(l.second.name + "\t" + formatValue(l.second.type, s.stack[l.first]));
+
+			QStringList	storage;
+			for(auto st: s.storage)
+			{
```
Comment (gavofyork) [inline mix/ClientModel.cpp]: unneeded `{`s.

File: mix/ClientModel.cpp [hunk 4]
```diff
@@ -339,18 +347,77 @@ void ClientModel::showDebuggerForTransaction(ExecutionResult const& _t)
...
+				}
+			}
+
+			//format solidity context values
+			QStringList	locals;
+			for(auto l: solLocals)
+				if (l.first < (int)s.stack.size())
+					locals.push_back(l.second.name + "\t" + formatValue(l.second.type, s.stack[l.first]));
+
+			QStringList	storage;
```
Comment (gavofyork) [inline mix/ClientModel.cpp]: tab in the middle of a line!?

File: mix/ClientModel.cpp [hunk 5]
```diff
@@ -339,18 +347,77 @@ void ClientModel::showDebuggerForTransaction(ExecutionResult const& _t)
...
+				}
+				else if (!returnStack.empty() && instructionIndex == returnStack.back())
+				{
+					returnStack.pop_back();
+					solCallStack.pop_back();
+				}
+			}
+
+			//format solidity context values
+			QStringList	locals;
```
Comment (gavofyork) [inline mix/ClientModel.cpp]: tab in the middle of a line!?

File: mix/CodeModel.cpp [hunk 6]
```diff
@@ -43,6 +46,91 @@ using namespace dev::mix;
 const std::set<std::string> c_predefinedContracts =
 	{ "Config", "Coin", "CoinReg", "coin", "service", "owned", "mortal", "NameReg", "named", "std", "configUser" };
 
+
+namespace
+{
+	using namespace dev::solidity;
```
Comment (gavofyork) [inline mix/CodeModel.cpp]: should be no additional indent level for namespace

File: mix/CodeModel.cpp [hunk 7]
```diff
@@ -43,6 +46,91 @@ using namespace dev::mix;
...
+	class CollectDeclarationsVisitor: public ASTConstVisitor
+	{
+	public:
+		CollectDeclarationsVisitor(QHash<LocationPair, QString>* _functions, QHash<LocationPair, SolidityDeclaration>* _locals, QHash<unsigned, SolidityDeclaration>* _storage):
+		m_functions(_functions), m_locals(_locals), m_storage(_storage), m_functionScope(false), m_storageSlot(0) {}
+	private:
```
Comment (gavofyork) [inline mix/CodeModel.cpp]: blank line before `private:`

File: mix/CodeModel.cpp [hunk 8]
```diff
@@ -43,6 +46,91 @@ using namespace dev::mix;
...
+		QHash<LocationPair, QString>* m_functions;
+		QHash<LocationPair, SolidityDeclaration>* m_locals;
+		QHash<unsigned, SolidityDeclaration>* m_storage;
+		bool m_functionScope;
+		uint m_storageSlot;
```
Comment (gavofyork) [inline mix/CodeModel.cpp]: private fields go beneath private methods.

**Comment by gavofyork** [comment]
a load of style stuff - also this seem to have substantial footprint in mix and solidity - is that intentional?

**Comment by CJentzsch** [comment]
What has happend to my PR? All this mix/solidity stuff is not from me and should not be in this PR.

**Comment by debris** [comment]
merge with lastest develop, and see it helps... probably you based your changes on rebased branch and github sometimes can't handle this.

**Comment by CJentzsch** [comment]
I did rebase on latest develop - this did work

**Pull request merged**